src/testing/basic_test.rs

```diff
@@ -1,390 +1,230 @@
+use async_std::prelude::FutureExt;
+use hotshot::types::SignatureKey;
+use hotshot_types::{simple_vote::QuorumData, traits::block_contents::vid_commitment};
 
-    use crate::builder_state::{
-        BuiltFromProposedBlock, DaProposalMessage, DecideMessage, QuorumProposalMessage,
-        RequestMessage, TransactionSource,
-    };
-    use crate::service::{broadcast_channels, handle_received_txns, GlobalState};
-    use crate::utils::BuilderStateId;
-    use async_lock::RwLock;
-    use async_std::task;
-    use committable::{Commitment, CommitmentBoundsArkless, Committable};
-    use std::sync::Arc;
-    use std::time::Duration;
+use hotshot_example_types::{
+    block_types::{TestBlockHeader, TestBlockPayload, TestMetadata, TestTransaction},
+    state_types::{TestInstanceState, TestValidatedState},
+};
 
-    use serde::{Deserialize, Serialize};
-    /// This test simulates multiple builder states receiving messages from the channels and processing them
-    #[async_std::test]
-    //#[instrument]
-    async fn test_builder() {
-        async_compatibility_layer::logging::setup_logging();
-        async_compatibility_layer::logging::setup_backtrace();
-        tracing::info!("Testing the builder core with multiple messages from the channels");
-        #[derive(
-            Copy,
-            Clone,
-            Debug,
-            Default,
-            Hash,
-            PartialEq,
-            Eq,
-            PartialOrd,
-            Ord,
-            Serialize,
-            Deserialize,
-        )]
-        struct TestTypes;
-        impl NodeType for TestTypes {
-            type Time = ViewNumber;
-            type BlockHeader = TestBlockHeader;
-            type BlockPayload = TestBlockPayload;
-            type SignatureKey = BLSPubKey;
-            type Transaction = TestTransaction;
-            type ValidatedState = TestValidatedState;
-            type InstanceState = TestInstanceState;
-            type Membership = GeneralStaticCommittee<TestTypes, Self::SignatureKey>;
-            type BuilderSignatureKey = BuilderKey;
-            type AuctionResult = TestAuctionResult;
+use crate::utils::BuilderStateId;
+use crate::{
+    builder_state::{DaProposalMessage, QuorumProposalMessage, RequestMessage, TransactionSource},
+    testing::TestTypes,
+};
+use crate::{service::handle_received_txns, testing::start_builder_state};
+use committable::Committable;
+use std::sync::Arc;
+use std::time::Duration;
+
+/// This test simulates multiple builder states receiving messages from the channels and processing them
+#[async_std::test]
+async fn test_builder() {
+    async_compatibility_layer::logging::setup_logging();
+    async_compatibility_layer::logging::setup_backtrace();
+    tracing::info!("Testing the builder core with multiple messages from the channels");
+
+    // Number of views to simulate
+    const NUM_ROUNDS: usize = 5;
+    // Number of transactions to submit per round
+    const NUM_TXNS_PER_ROUND: usize = 5;
```

Comment:
Can we make this different from `NUM_ROUNDS`, like 4? It's not a big deal for now, but it may make future debugging easier.